bin/collator/src/benchmarking.rs

```diff
@@ -0,0 +1,512 @@
+        );
+
+        let payload = runtime::SignedPayload::from_raw(
+            call.clone(),
+            extra.clone(),
+            (
+                runtime::VERSION.spec_version,
+                runtime::VERSION.transaction_version,
+                genesis.clone(),
+                genesis,
+                (),
+                (),
+                (),
+            ),
+        );
+
+        let signature = payload.using_encoded(|p| acc.sign(p));
+        runtime::UncheckedExtrinsic::new_signed(
+            call,
+            sp_runtime::AccountId32::from(acc.public()).into(),
+            runtime::Signature::Sr25519(signature.clone()),
+            extra,
+        )
+        .into()
+    }
+}
+
+/// Provides the existential deposit that is only needed for benchmarking.
+pub trait ExistentialDepositProvider {
+    /// Returns the existential deposit.
+    fn existential_deposit(&self) -> Balance;
+}
+
+impl<RuntimeApi, Executor> ExistentialDepositProvider
+    for TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>
+where
+    RuntimeApi: ConstructRuntimeApi<Block, TFullClient<Block, RuntimeApi, NativeElseWasmExecutor<Executor>>>
+        + Send
+        + Sync
+        + 'static,
+    Executor: sc_executor::NativeExecutionDispatch + 'static,
+{
+    fn existential_deposit(&self) -> Balance {
+        with_runtime! {
+            self,
+            runtime::ExistentialDeposit::get()
+        }
+    }
+}
+
+/// Generates inherent data for benchmarking Astar, Shiden and Shibuya.
+///
+/// Not to be used outside of benchmarking since it returns mocked values.
+pub fn local_benchmark_inherent_data(
+) -> std::result::Result<sp_inherents::InherentData, sp_inherents::Error> {
+    use sp_inherents::InherentDataProvider;
+    let mut inherent_data = sp_inherents::InherentData::new();
+
+    // Assume that all runtimes have the `timestamp` pallet.
+    let d = std::time::Duration::from_millis(0);
+    let timestamp = sp_timestamp::InherentDataProvider::new(d.into());
+    timestamp.provide_inherent_data(&mut inherent_data)?;
+
+    Ok(inherent_data)
+}
+
+/// Generates inherent data for benchmarking Astar, Shiden and Shibuya.
+///
+/// Not to be used outside of benchmarking since it returns mocked values.
+pub fn para_benchmark_inherent_data(
+) -> std::result::Result<sp_inherents::InherentData, sp_inherents::Error> {
+    use sp_inherents::InherentDataProvider;
+    let mut inherent_data = sp_inherents::InherentData::new();
+
+    // Assume that all runtimes have the `timestamp` pallet.
+    let d = std::time::Duration::from_millis(0);
+    let timestamp = sp_timestamp::InherentDataProvider::new(d.into());
+    timestamp.provide_inherent_data(&mut inherent_data)?;
+
+    let sproof_builder = RelayStateSproofBuilder::default();
+    let (relay_parent_storage_root, relay_chain_state) = sproof_builder.into_state_root_and_proof();
+    let validation_data = PersistedValidationData {
+        relay_parent_number: 1,
```

Comment:
`relay_parent_number` need to be more than 0 for benchmarking, otherwise panics at https://github.com/paritytech/cumulus/blob/3b3b6d2f8420824c52ebf7385465a919c96f82c4/pallets/parachain-system/src/lib.rs#L122